packages/langchain-graph-retriever/src/langchain_graph_retriever/adapters/astra.py

```diff
@@ -395,20 +320,28 @@ def _execute_query(
         include_similarity = None
         if not (sort or {}).keys().isdisjoint({"$vector", "$vectorize"}):
             include_similarity = True
-        hits = astra_env.collection.find(
-            filter=query,
-            projection=self.vector_store.document_codec.full_projection,
-            limit=limit,
-            include_sort_vector=True,
-            include_similarity=include_similarity,
-            sort=sort,
-        )
 
-        return [content for hit in hits if (content := self._decode_hit(hit))]
+        results: dict[str, Content] = {}
+        for filter in filters:
+            # TODO: Look at a thread-pool for this.
+            hits = astra_env.collection.find(
+                filter=filter,
+                projection=self.vector_store.document_codec.full_projection,
+                limit=limit,
+                include_sort_vector=True,
+                include_similarity=include_similarity,
+                sort=sort,
+            )
 
-    async def _aexecute_query(
+            for hit in hits:
+                if hit["_id"] not in results and (content := self._decode_hit(hit)) is not None:
+                    results[content.id] = content
+
+        return list(results.values())
+
+    async def _aexecute_queries(
         self,
-        query: dict[str, Any] | None = None,
+        filters: Iterator[dict[str, Any]],
```

Comment:
Execution methods changed to take an iterator of filters, run all of them, and combine them. They only decode new IDs.